tests/ui/closures/2229_closure_analysis/issue-118144.rs

```diff
@@ -0,0 +1,14 @@
+// Regression test for ICE #118144
+
+struct V(i32);
+
+fn func(func_arg: &mut V) {
+    || {
+        let x;
+        V(x) = func_arg; //~ ERROR: mismatched types
```

Comment:
Is there no way to destructure-assign a reference??

Reply:
If you use `let V(x) = ...` then the ICE goes away. See the second last paragraph on this comment on the issue: https://github.com/rust-lang/rust/issues/118144#issuecomment-1837467068.

I'll add a comment explaining this in the test code.